### PR TITLE
MODSOURAMN-1063 - Update RecordProcessingLogDto to contain incoming record id

### PR DESCRIPTION
**Pull request by VRohach**
## Purpose
Added incomingRecordId in https://github.com/folio-org/data-import-raml-storage/blob/master/schemas/dto/recordProcessingLogDto.json schema. SourceRecordId will still correspond to SRS record id, but may be empty if record is not saved in srs, therefore - remove sourceRecordId from "required" section
Checked specifics of Invoice Lines display. What 'relatedInvoiceLineInfo' refers to and add to ProcessingLogDto if needed
Made changes in MetadataProvider services and dao to populate incomingRecordId in ProcessingLogDto
Made changes in MetadataProvider for /metadata-provider/jobLogEntries/{jobExecutionId} to return collection of ProcessingLogDto. jobLogEntryDtoCollection schema should be changed or new collection schema is added
[Remove https://github.com/folio-org/data-import-raml-storage/blob/master/schemas/dto/jobLogEntryDto.json](https://github.com/folio-org/data-import-raml-storage/blob/master/schemas/dto/jobLogEntryDto.json)
Updated tests

## Approach
get_job_log_entries function improved to be able to fill the "RecordProcessingLogDto"-entity in collection.

**Depends on:** https://github.com/folio-org/data-import-raml-storage/pull/304

## Checklist
- [x] I have updated NEWS.md.
- [ ] I have added javadocs to new methods.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation e.g. README.md.
- [ ] I have ran karate tests against this feature.

### CODE REVIEW

**Comment by VRohach** [comment]
Schema changes will be done in the separate PR after this implementation will be tested. For testing purpose changes in schema is provided in this way in .raml-file: 
" recordProcessingLogDto: !include tmpSchemas/recordProcessingLogDto.json
  recordProcessingLogDtoCollection: !include tmpSchemas/recordProcessingLogDtoCollection.json"

**Comment by VRohach** [comment]
> Schema changes will be done in the separate PR after this implementation will be tested. For testing purposes changes in schema is provided in this way in .raml-file: " recordProcessingLogDto: !include tmpSchemas/recordProcessingLogDto.json recordProcessingLogDtoCollection: !include tmpSchemas/recordProcessingLogDtoCollection.json"

**This should be done when merging to master-branch!!!**

File: mod-source-record-manager-server/src/main/java/org/folio/dao/JournalRecordDaoImpl.java [hunk 1]
```diff
@@ -137,6 +136,10 @@ public class JournalRecordDaoImpl implements JournalRecordDao {
   private static final Logger LOGGER = LogManager.getLogger();
   public static final String SOURCE_RECORD_ENTITY_TYPE = "source_record_entity_type";
   public static final String ORDER_ENTITY_ID = "order_entity_id";
+  public static final String SOURCE_ENTITY_ERROR = "source_entity_error";
```
Comment (yaroslav-epam) [inline mod-source-record-manager-server/src/main/java/org/folio/dao/JournalRecordDaoImpl.java:139]: looks like `HOLDINGS_ENTITY_HRID` and `SOURCE_ENTITY_ERROR` constants are already defined in `JournalRecordsColumns`

**Review by KaterynaSenchenko — APPROVED** [review]

File: mod-source-record-manager-server/src/test/java/org/folio/rest/impl/metadataProvider/MetaDataProviderRecordProcessingLogCollectionAPITest.java [hunk 2]
```diff
@@ -0,0 +1,1634 @@
...
+  public void shouldNotReturnMarcBibRecordsWhenInstanceDiscarderRetrievingWithErrorsOnlyParam(TestContext context) {
+    Async async = context.async();
+    JobExecution createdJobExecution = constructAndPostInitJobExecutionRqDto(1).getJobExecutions().get(0);
+
+    String sourceRecordId1 = UUID.randomUUID().toString();
+    String sourceRecordId2 = UUID.randomUUID().toString();
+    String sourceRecordId3 = UUID.randomUUID().toString();
+
+    Future<JournalRecord> future = Future.succeededFuture()
+      /* .compose(v -> createJournalRecord(createdJobExecution.getId(), sourceRecordId1, null, null, null, 1, CREATE, MARC_BIBLIOGRAPHIC, COMPLETED, null, null))
```
Comment (yaroslav-epam) [inline mod-source-record-manager-server/src/test/java/org/folio/rest/impl/metadataProvider/MetaDataProviderRecordProcessingLogCollectionAPITest.java:1166]: commented out code

**Review by yaroslav-epam — APPROVED** [review]

**Review by JavokhirAbdullayev — APPROVED** [review]

**Pull request merged**